pkg/proxy/iptables/proxier.go

```diff
@@ -1292,13 +1394,24 @@ func (proxier *Proxier) syncProxyRules() {
 	natLines := append(natChains.Bytes(), natRules.Bytes()...)
 	lines := append(filterLines, natLines...)
 
-	glog.V(3).Infof("Restoring iptables rules: %s", lines)
-	err = proxier.iptables.RestoreAll(lines, utiliptables.NoFlushTables, utiliptables.RestoreCounters)
+	filterSame, err := sameIptablesRuleset(utiliptables.TableFilter, filterLines, iptablesFilterSaveRaw)
 	if err != nil {
-		glog.Errorf("Failed to execute iptables-restore: %v", err)
-		// Revert new local ports.
-		revertPorts(replacementPortsMap, proxier.portsMap)
-		return
+		glog.Warningf("failed to parse iptables filter rules: %v", err)
```

Comment:
The proxy testcases should pick up changes to the ruleset though, so it should fail pretty early if syncProxyRules() gets changed.  And it's at least designed to fail safe, as in it shouldn't change existing behavior if things don't work, it'll just always sync and log a message.

Ideally instead of adding textual rules we'd have a more structured iptables module that would use the vishvananda/netlink to read/write stuff and thus be able to more directly compare instead of screen-scraping, but that's a ton of work...

If you really want it I can try the EnsureRule thing and add hostports, userspace proxy, and other testcases too.